content/docs/userspace/gall-guide/7-sur-and-marks.md

```diff
@@ -0,0 +1,375 @@
+    [cards this]
+  ++  handle-poke
+    |=  =action:todo
+    ^-  (quip card _state)
+    ?-    -.action
+        %add
+      :_  state(tasks (~(put by tasks) now.bowl [name.action %.n]))
+      :~  [%give %fact ~[/updates] %noun !>(`update:todo`[%add now.bowl name.action])]
+      ==
+    ::
+        %del
+      :_  state(tasks (~(del by tasks) id.action))
+      :~  [%give %fact ~[/updates] %noun !>(`update:todo`action)]
+      ==
+    ::
+        %toggle
+      :_  state(tasks (~(jab by tasks) id.action |=(=task task(done !done.task))))
+      :~  [%give %fact ~[/updates] %noun !>(`update:todo`action)]
+      ==
+    ::
+        %rename
+      :_  state(tasks (~(jab by tasks) id.action |=(=task task(name name.action))))
+      :~  [%give %fact ~[/updates] %noun !>(`update:todo`action)]
+      ==
+    ==
+  --
+```
+
+Let's break this down a bit. Firstly, our `on-poke` arm includes a
+[barket](/docs/hoon/reference/rune/bar#-barket) (`|^`) rune. Barket creates a
+core with a `$` arm that's computed immediately. We extract the `vase` to the
+`action:todo` type and immediately pass it to the `handle-poke` arm of the core
+created with the barket. This `handle-poke` arm tests what kind of `action` it's
+received by checking its head. It then updates the state, and also sends an
+update to subscribers, as appropriate. Don't worry too much about the `%give`
+`card` for now - we'll cover subscriptions in the next lesson.
+
+Notice that the `handle-poke` arm produces a `(quip card _state)` rather than
+`(quip card _this)`. The call to `handle-poke` is also part of the following
+expression:
+
+```hoon
+=^  cards  state
+  (handle-poke !<(action:todo vase))
+[cards this]
+```
+
+The [tisket](/docs/hoon/reference/rune/tis#-tisket) (`=^`) expression takes two
+arguments: A new named noun to pin to the subject (`cards` in this case), and an
+existing wing of the subject to modify (`state` in this case). Since
+`handle-poke` produces `(quip card _state)`, we're saving the `card`s it
+produces to `cards` and replacing the existing `state` with its new one.
+Finally, we produce `[cards this]`, where `this` will now contain the modified
+`state`. The `[cards this]` is a `(quip card _this)`, which our `on-poke` arm is
+expected to produce.
+
+This might seem a little convoluted, but it's a common pattern we do for two
+reasons. Firstly, it's not ideal to be passing around the entire `this` agent
+core - it's much tidier just passing around the `state`, until you actually want
+to return it to Gall. Secondly, It's much easier to read when the poke handling
+logic is separated into its own arm. This is a fairly simple example but if your
+agent is more complex, handling multiple `mark`s and containing additional logic
+before it gets to the actual contents of the `vase`, structuring things this way
+can be useful.
+
+You can of course structure your `on-poke` arm differently than we've done
+here - we're just demonstrating a typical pattern.
+
+## `mark` files
```

Comment:
This section uses mark with and without backtics frequently. Is there a pattern to it? I can think of reasons to vary the usage, e.g. "mark" when referring to a specific mark and `mark` when referring to the general notion (or vice versa). I am generally of the opinion that very frequently used terms in Urbit don't need backtics (which is why I've been removing them from door everywhere), but marks are sort of a borderline case I think so this is tricky.